### ISSUE TITLE
Build failure on msvc-10.0 and earlier

### ISSUE THREAD

**Issue by pdimov** [boostorg/sync#7]
```
compile-c-c++ ..\..\bin.v2\libs\sync\build\msvc-10.0\debug\threadapi-win32\threa
ding-multi\tss_windows.obj
tss_windows.cpp
C:\Projects\boost-git\boost\boost/sync/exceptions/resource_error.hpp(40) : error
 C2440: 'static_cast' : cannot convert from 'boost::exception_detail::error_info
_injector<T>' to 'int'
        with
        [
            T=boost::sync::resource_error
        ]
        No user-defined-conversion operator available that can perform this conv
ersion, or the operator cannot be called
```

**Comment by Lastique** [comment]
Should probably be fixed after c72891d.

**Comment by pdimov** [comment]
Confirmed, thanks.
